dawn/renderer/components/Editor.js

```diff
@@ -182,6 +187,19 @@ class Editor extends React.Component {
       });
       throw err;
     });
+    ipcRenderer.send('NOTIFY_UPLOAD');
+    while (this.notificationHold === 1) {
```

Comment:
Is this intentionally blocking the thread until notificationHold is not 1? Consider using a Promise or something else instead.